### DISCUSSION TITLE
How to update plugins for the CLI next to use config files?

### DISCUSSION THREAD

**Discussion by JillieBeanSim** [zowe/zowe-cli#1112] [Q&A]
I am working to update our Zowe CLI plugin's next to recognize the config files.  I have the `zowe config init` command adding our plugin with default property values but when I run a command I get this error:
```
Command Error:
An error occurred collecting all configurations from the profile root directory "C:\Users\AUser\.zowe\profiles". Please supply the configurations on the profile manager constructor parameters OR initialize the profile manager environment. Details: Profile IO Error: A Zowe V1 profile operation was attempted with a Zowe V2 configuration in use.
```
How do I go about resolving this issue to fulfill the Zowe V2 configuration?

**Comment by t1m0thyj** [comment]
This error means that a v1 profile API was called in an environment where v2 profiles are in use.

Does your plugin call the CliProfileManager at all or perform any reading of profiles from disk?

**Reply by JillieBeanSim** [reply]
On the CLI side it does not, before with v1 it was all handle by Zowe CLI.  I haven't come across any documentation on this for plugins .

**Reply by JillieBeanSim** [reply]
@t1m0thyj After reading from disk in our Base Handler I was able to get a default profile to work with the config file.  Unfortunately we cannot get the `IHandlerParameters` of a profile that isn't default when using the equivalent of `--zosmf-p`.  I get this error when adding the tag and other profile name
![image](https://user-images.githubusercontent.com/49491949/130275627-c4097703-57a4-4f19-9966-4ffc513a35dd.png)

**Reply by gejohnston** [reply]
Is this a similar problem to the one identified in the slack channel "Open Mainframe Project : zowe-cli" ?

Both seem to want to access a profile other than the primary profile of the plugin. I pasted the text below for convenient reading.
________________________________________
Jakub Sobek  9:56 AM
Hi, how can I use SSH and zosmf APIs in a single command?
Something likes this:
const zosmfProfile = params.profiles.get("zosmf");
const zosmfSession = ZosmfSession.createBasicZosmfSession(zosmfProfile);
const sshfProfile = params.profiles.get("ssh");
const sshSession = SshSession.createBasicSshSession(sshProfile);
Above methods are deprecated and the new way createSessCfgFromArgs loads all required profiles and rewrites variables of the same name(host,port atd.)
I didn't find another way to do this.

Gene Johnston  10:01 AM
Are you working in a CLI app or a VSCode type of app (like Zowe Explorer)?

Jakub Sobek  10:02 AM
its a Zowe CLI plugin

Gene Johnston  10:10 AM
In a CLI app, we advise that you do not directly read the profile. The property values from the command line, env, and profiles are automatically resolved and supplied in the params:IHandlerParameters parameter of your handler. The medium article https://medium.com/zowe/an-api-to-simplify-retrieval-of-zowe-profile-information-a1293656de21 mostly does not apply to your situation, but the section titled "Which apps should use the ProfileInfo API ?" describes this a bit better than I just did. The CreateSesscfgFromArgs function takes advantage of the fact that all the parameters have already been resolved according to the Zowe order-of-precedence.

Jakub Sobek  10:22 AM
The problem I'm facing is that ssh profile and zosmf profile have variables of the same name.
Both profiles have port but only one will be supplied with params
so I cant use CreateSesscfgFromArgs to create sessions for both ssh and zosmf, because both will point to the same port.

Gene Johnston  10:39 AM
Ah, I see the problem. I cannot think of an immediate answer. Deprecated functions will continue to work in V1. That may be a short-term solution. However, most will disappear in V2. A GitHub issue may be required to get a solution implemented for this problem.

Jakub Sobek  10:43 AM
Ok, thank you for the help. I wasn't sure if I was missing something and I will create the issue.
